### PR TITLE
Improve styles for facet dialog

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Since we refactored the facet dialog we haven't done a lot on the styling for the mobile experience. Moreover, we found a bug regarding the scrolling behaviour of the underlying body element.

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts [hunk 1]
```diff
@@ -21,10 +23,21 @@ import { FacetService } from '../services/facet.service';
...
+  private _isDialog: boolean;
   /**
    * Indicates that the facet navigation is rendered in dialog.
    */
-  @Input() isDialog: boolean;
+  @Input()
+  set isDialog(value: boolean) {
+    this._isDialog = value;
+    if (value) {
+      this.renderer.addClass(document.body, 'modal-open');
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts:34]: Should we also add `aria-modal="true"`?

Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts:34]: Can we make some configurable classes here?

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss [hunk 2]
```diff
@@ -1,35 +1,89 @@
...
+  align-items: center;
+
+  > .inner {
     transform: translate3d(0, 100%, 0);
+  }
+}
 
-    &.active {
-      transform: translate3d(0, 0, 0);
+%cx-overlay {
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/components/product/list/_facet-list.scss:42]: Is this used?

File: projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts:34]: I don't mind leaving this to the next round of accessibility that is planned post 2.0

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss (hunk 2 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontstyles/scss/components/product/list/_facet-list.scss:42]: This is causing the animation effect. Once the dialog becomes active, the transform kick in and neutralises to the zero position. The original position can be left, bottom, etc.

**Review by LTiger14 — DISMISSED** [review]

**Pull request merged**